### ISSUE TITLE
Add "multi-iteration gradient calculation" functionality

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#3575]
Priority: low. Only benefit is for memory constrained training situations.

Idea: Caffe has an "iter_size" parameter.
What it does: iter_size = X means "accumulate gradients for X separate minibatches, and then do one parameter update based on that"

Consequently, we have multiple forward/backward passes per parameter update; mathematically it should be identical to single minibatch case in terms of the trained net, though will use less memory (for the activations and backprop gradients).

**Comment by raver119** [comment]
Is this still relevant?
